chain/bitcoind_events.go

```diff
@@ -0,0 +1,321 @@
+package chain
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightninglabs/gozmq"
+)
+
+// BitcoindEvents is the interface that must be satisfied by any type that
+// serves bitcoind block and transactions events.
+type BitcoindEvents interface {
+	Start() error
```

Comment:
Hmm, fwiw this interface as defined, doesn't really say much (anything w/ a start + stop method). What if we modified the interface to instead define two methods (in addition to these) which each return a channel to listen on for events? 

So something like this: 
```go
type BitcoindEvents interface {
    TxNotifications() <-chan *wire.MsgTx
    BlockNotifications() <-chan *wire.MsgBlock
}
```

I imagine this interface would also make it easier to write generic unit tests as well.